Composer/packages/client/src/recoilModel/atoms/appState.ts

```diff
@@ -152,6 +153,11 @@ export const boilerplateVersionState = atom<BoilerplateVersion>({
   },
 });
 
+export const notificationsState = atom<Notification[]>({
```

Comment:
Have you considered using an atomFamily?

Reply:
If there are multiple notifications showing and user deletes one of them or a new notification gets added to the list, the entire list gets re-rendered. I suggest doing an atom to keep the notificationIds and an atomFamily for the notifications content using that id, then use React.memo on the notification component to prevent re-rendering if that atomFamily hasn't changed (aka non of the existing notifications gets re-rendered when a new notification comes in or one gets deleted)